lightning/src/util/chacha20.rs

```diff
@@ -572,4 +591,32 @@ mod test {
 			assert_eq!(output, tv.keystream);
 		}
 	}
+
+	#[test]
+	fn get_single_block() {
+		// Test that `get_single_block` (which takes a 16-byte nonce) is equivalent to getting a block
+		// using a 12-byte nonce, with the block starting at the counter offset given by the remaining 4
+		// bytes.
+		let key = [
+			0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+			0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+			0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17,
+			0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f,
+		];
+		let nonce_16bytes = [
+			0x00, 0x01, 0x02, 0x03, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b
+		];
+		let mut counter_pos = [0; 4];
+		counter_pos.copy_from_slice(&nonce_16bytes[..4]);
+		let nonce_12bytes = &nonce_16bytes[4..];
+
+		// Initialize a ChaCha20 instance with its counter starting at 0.
+		let mut chacha20 = ChaCha20::new(&key, nonce_12bytes);
+		// Seek its counter to the block at counter_pos.
+		chacha20.seek_to_block(u32::from_le_bytes(counter_pos).try_into().unwrap());
```

Comment:
Hmm, `from_le_bytes` takes a `[u8; 4]`, not a reference so the compiler complains -- am i missing something?

Reply:
Should work as if you move the `try_into` immediately after the slice instead of outside `u32::from_le_bytes`:

```rust
let counter_pos = &nonce_16bytes[..4];
chacha20.seek_to_block(u32::from_le_bytes(counter_pos.try_into().unwrap()));
```